blockchain/chain.go

```diff
@@ -774,6 +801,10 @@ func (b *BlockChain) connectBlock(node *blockNode, block, parent *dcrutil.Block,
 		return err
 	}
 
+	blockSize := uint64(block.MsgBlock().Header.Size)
+	state := newBestState(node, blockSize, numTxns, curTotalTxns+numTxns,
```

Comment:
This is wrong.  It can't be set before the database is updated.  It needs to stay where it was.